lisa/features/non_ssh_executor.py

```diff
@@ -0,0 +1,52 @@
+from typing import List
+
+from lisa.feature import Feature
+from lisa.features.serial_console import SerialConsole
+
+
+class NonSshExecutor(Feature):
+    """
+    NonSshExecutor is used to run commands on the node when SSH is not available.
+    Lisa by default uses SSH for connection, but this feature provides an alternative
+    execution method for scenarios where SSH connectivity is not possible or desired.
+    """
+
+    @classmethod
+    def name(cls) -> str:
+        return "NonSshExecutor"
+
+    def enabled(self) -> bool:
+        return True
+
+    def execute(self, commands: List[str]) -> List[str]:
+        """
+        Executes a list of commands on the node and returns their outputs.
+
+        :param commands: A list of shell commands to execute.
+        :return: A string containing the output of the executed commands.
+        """
+        out = []
+
+        serial_console = self._node.features[SerialConsole]
+        if not serial_console.enabled():
```

Comment:
use `self._node.features.is_supported` to check the feature is available or not before `self._node.features[SerialConsole]`